pulsar-functions/proto/src/main/proto/Function.proto

```diff
@@ -100,12 +100,38 @@ message ConsumerSpec {
     ReceiverQueueSize receiverQueueSize = 4;
     map<string, string> schemaProperties = 5;
     map<string, string> consumerProperties = 6;
+    CryptoSpec cryptoSpec = 7;
 }
 
 message ProducerSpec {
     int32 maxPendingMessages = 1;
     int32 maxPendingMessagesAcrossPartitions = 2;
     bool useThreadLocalProducers = 3;
+    CryptoSpec cryptoSpec = 4;
+}
+
+message CryptoSpec {
+    enum FailureAction {
+        FAIL = 0;
+
+        DISCARD = 1;
+        CONSUME = 2;
+
+        SEND = 10;
```

Comment:
Reserve some single digit values just in case there're new actions added for the consumer failure.

Logically it's more clear. FAIL or 0 is the default behavior; 1~9 is for consumer; 10 and above is for producer.